plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/WarningChecksPublisherITest.java

```diff
@@ -0,0 +1,104 @@
+import io.jenkins.plugins.checks.api.ChecksStatus;
+
+import static io.jenkins.plugins.analysis.core.assertions.Assertions.*;
+
+public class WarningChecksPublisherITest extends IntegrationTestWithJenkinsPerTest {
+    private static final String OLD_REPORT_FILE = "checkstyle.xml";
+    private static final String NEW_REPORT_FILE = "checkstyle1.xml";
+
+    /**
+     * Verifies that {@link WarningChecksPublisher} constructs the {@link ChecksDetails} correctly
+     * with only new issues.
+     */
+    @Test
+    public void shouldCreateChecksDetailsWithNewIssuesAsAnnotations() {
+        FreeStyleProject project = createFreeStyleProjectWithWorkspaceFiles(OLD_REPORT_FILE);
+        enableCheckStyleWarnings(project);
+
+        // trigger a build using the report file with 4 issues
+        Run<?, ?> run = buildSuccessfully(project);
+        assertThat(getAnalysisResult(run))
+                .hasTotalSize(4)
+                .hasNewSize(0);
+
+        // trigger a build using the report file with 2 new issues
+        copyMultipleFilesToWorkspaceWithSuffix(project, NEW_REPORT_FILE);
+        run = buildSuccessfully(project);
+        assertThat(getAnalysisResult(run))
+                .hasTotalSize(6)
+                .hasNewSize(2);
+
+        // extract details from result
+        WarningChecksPublisher publisher = new WarningChecksPublisher(getResultAction(run));
+        ChecksDetails details = publisher.extractChecksDetails();
+
+        // verify extracted details
+        ChecksDetails expectedDetails = createChecksDetailsBasedOnReportFile();
+        assertThat(details)
+                .usingRecursiveComparison()
+                .ignoringFields("detailsURL")
+                .isEqualTo(expectedDetails);
+    }
+
+    private ChecksDetails createChecksDetailsBasedOnReportFile() {
+        ChecksDetailsBuilder builder = new ChecksDetailsBuilder()
+                .withName("CheckStyle")
+                .withStatus(ChecksStatus.COMPLETED)
+                .withConclusion(ChecksConclusion.SUCCESS)
+                .withDetailsURL("http://localhost:39121/jenkins/job/test0/2/");
+
+
+        ChecksOutput output = new ChecksOutputBuilder()
+                .withTitle("CheckStyle Warnings")
+                .withSummary("## 6 issues in total:\n"
+                        + "- ### 2 new issues\n"
+                        + "- ### 4 outstanding Issues\n"
+                        + "- ### 2 delta issues\n"
+                        + "- ### 0 fixed issues")
+                .withText("## Total Issue Statistics:\n* Error: 6\n* High: 0\n* Normal: 0\n* Low: 0\n"
+                        + "## New Issue Statistics:\n* Error: 2\n* High: 0\n* Normal: 0\n* Low: 0\n"
+                        + "## Delta Issue Statistics:\n* Error: 2\n* High: 0\n* Normal: 0\n* Low: 0\n")
+                .addAnnotation(new ChecksAnnotationBuilder()
+                        .withPath("X:/Build/Results/jobs/Maven/workspace/tasks/src/main/java/hudson/plugins"
```

Comment:
What are the requirements on GitHub side for the resolution of the filenames? Do they want relative paths starting from the repository root?

Reply:
Not officially documented by GitHub, but from what I have done, it's relative paths from the repository root.

Reply:
Ok, then using `getFileName` should work. (This is not always so easy since people sometimes design jobs using a complex directory layout. Then it is already hard for my plugin to refer to the correct source code location)